src/Transformer/CustomTransformer/CustomPropertyTransformerInterface.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AutoMapper\Transformer\CustomTransformer;
+
+/**
+ * This interface should be implemented to handle custom transformations for specific property,
+ * from a given source type to a given target type:.
+ *
+ * ```php
+ * final class UppercaseUserNameCustomTransformer implements CustomModelTransformerInterface
+ * {
+ *     public function supports(string $source, string $target, string $propertyName): bool
+ *     {
+ *         return $source === UserDTO::class && $target === User::class && $propertyName === 'name'
+ *     }
+ *
+ *     public function transform(mixed $input): string
+ *     {
+ *        assert($input instanceof UserDTO);
+ *
+ *         return strtoupper($input->name);
+ *     }
+ * }
+ * ```
+ *
+ * @experimental Interface contract will change once https://github.com/symfony/symfony/pull/52510 will be merged.
+ */
+interface CustomPropertyTransformerInterface extends CustomTransformerInterface
+{
+    /**
+     * @param 'array'|class-string $source
+     * @param 'array'|class-string $target
+     */
+    public function supports(string $source, string $target, string $propertyName): bool;
```

Comment:
Shouldn't it be an array for both `source` & `target` parameters ?

Reply:
since we're at "property level", we're using types from `MapperMetadata`'s source and target.
See https://github.com/jolicode/automapper/pull/10/files#diff-f42c4b3c8fc2137483ba833fa3b79ab8bd81932296df954c775997278f22a4bdR39

I don't think we can have an array here. Or... I've hooked at the wrong place for custom property transformer